src/main/java/org/candlepin/service/exception/subscription/SubscriptionServiceExceptionMapper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.exception.subscription;
+
+import org.xnap.commons.i18n.I18n;
+
+
+
+public class SubscriptionServiceExceptionMapper {
+    private SubscriptionServiceExceptionMapper() { };
+
+    public static String map(SubscriptionServiceException e, String dellAssetTag, String ownerKey,
+        I18n i18n) {
+        String message = "";
+        if (e instanceof SubscriptionActivationException) {
+            message = i18n.tr(
+                "No subscription was able to be activated with Dell service tag '{0}'.", dellAssetTag);
```

Comment:
Adding onto the comment above, this handler runs directly into the problem these mappers can hit without care. `SubscriptionActivationException` is pretty generic, but because of the state required to be passed in with the mapping function itself, there's an assumption that this only occurs when registering with an asset tag, and that is reflected in the output message. This limits the scope of the mapper to only the one or two places where tag-based activation is invoked, somewhat defeating the point of abstracting this logic out.